src/core/runner.ts

```diff
@@ -207,25 +207,29 @@ export default class Runner extends EventEmitter {
       data.error = error;
     } finally {
       data.url ??= driver.page.url();
-      if (screenshots) {
+      if (screenshots !== 'off') {
```

Comment:
Could we clean up this function and extract the screenshot portion to a separate function for readability?  `runStep` is getting quite large.

Reply:
Done.